app/src/main/java/com/example/sharingang/MainActivity.kt

```diff
@@ -78,17 +106,23 @@ class MainActivity : AppCompatActivity(R.layout.activity_main) {
             }
     }
 
+    // We need the parameter for the callback
     fun onRandomItem(item: MenuItem) {
+        selectRandomItem()
+    }
+
+    private fun selectRandomItem() {
         lifecycleScope.launch(Dispatchers.IO) {
             val allItems = itemRepository.getAll()
             if (allItems.isNotEmpty()) {
-                val random = allItems.random()
+
+                val random = itemRepository.getAll().random()
```

Comment:
Why do we need to `getAll()` again instead of keeping `allItems.random()`?

Reply:
Not sure I understand why either

Reply:
Mistake when merging :P